src/classes/PPlan_CannotDelete_TDTM.cls

```diff
@@ -46,18 +50,21 @@ public with sharing class PPlan_CannotDelete_TDTM extends TDTM_Runnable {
     ********************************************************************************************************/
     public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist,
             TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+        
+        // return if Program Plan Deletion is not enabled
+        if (!enabledPreventProgramPlanDeletion) {
+            return new DmlWrapper();
+        }
 
         Map<ID, SObject> oldmap = new Map<ID, SObject>(oldlist);
 
         if (triggerAction == TDTM_Runnable.Action.BeforeDelete) {
-
             for (Program_Plan__c programPlan : [SELECT ID,
                                                         (SELECT ID FROM Plan_Requirements__r LIMIT 1),
                                                         (SELECT ID FROM Program_Enrollments__r LIMIT 1)
                                                 FROM Program_Plan__c
                                                 WHERE ID IN :oldlist]) {
-
-                if(programPlan.Plan_Requirements__r.size() > 0
+                if (programPlan.Plan_Requirements__r.size() > 0
```

Comment:
This is more of a stylistic thought, but have we considered using List.isEmpty() and ! List.isEmpty() instead of size > 0? It does deviate from our prior code, but it feels more human-readable in how it is being explicit about what we are trying to have the code do.

Reply:
I usually prefer isEmpty() calls, however I haven't in the past really ever planted a flag anywhere. I like that we are thinking about it. The only thing I've noticed in the past is when calling !Somelist.isEmpty() that it is sometimes easy for eyes to miss the "!" -- all that said though, the context when reviewing said code block helps ensure that you're looking for it more closely when you see the isEmpty()

Reply:
Another perfectly valid way to make it more readable would be List.isEmpty() == false, which may be even more readable!